File: Software Technologies/Exam Preparation II/Exam-Prep-2-IMDB/Java Skeleton/src/main/java/imdb/entity/Film.java
package imdb.entity;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "films")
public class Film {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private String name;
    @Column
    private String genre;
    @Column
    private String director;
    @Column
    private Integer year;

    public Film(){
    }

    public Film(String name, String genre, String director, Integer year) {
        this.name = name;
        this.genre = genre;
        this.director = director;
        this.year = year;
    }

    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getGenre() {
        return genre;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }

    public String getDirector() {
        return director;
    }

    public void setDirector(String director) {
        this.director = director;
    }

    public Integer getYear() {
        return year;
    }

    public void setYear(Integer year) {
        this.year = year;
    }
}

File: Software Technologies/Exam Preparation II/Exam-Prep-2-IMDB/JavaScript Skeleton/models/film.js
const Sequelize = require('sequelize');

module.exports = function(sequelize){
    let Film = sequelize.define('Film', {
        name:{
            type: Sequelize.TEXT,
            allowNull: false,
            required: true
        },
        genre:{
            type: Sequelize.TEXT,
            allowNull: false,
            required: true
        },
        director:{
            type: Sequelize.TEXT,
            allowNull: false,
            required: true
        },
        year:{
            type: Sequelize.INTEGER,
            allowNull: false,
            required: true
        }
    }, {
        timestamps:false
    });


    return Film;
};